### PR TITLE
refactor: Add secret to metadata

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Refactoring

## Description

Added Secret wrapping to Metadata field.

## Motivation and Context

Added secret wrapping to metadata since it could have some sensitive info as well.

## How did you test it?

Manual.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -844,7 +841,7 @@ pub struct PaymentsResponse {
     pub billing: Option<Address>,
     /// You can specify up to 50 keys, with key names up to 40 characters long and values up to 500 characters long. Metadata is useful for storing additional, structured information on an object.
     #[schema(value_type = Option<Object>)]
-    pub metadata: Option<serde_json::Value>,
+    pub metadata: Option<Secret<serde_json::Value>>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:844]:
```suggestion
    pub metadata: Option<pii::SecretSerdeValue>,
```

File: crates/router/src/compatibility/stripe/payment_intents/types.rs [hunk 2]
```diff
@@ -63,7 +63,7 @@ pub struct StripePaymentMethodData {
     pub billing_details: Option<StripeBillingDetails>,
     #[serde(flatten)]
     pub payment_method_details: Option<StripePaymentMethodDetails>, // enum
-    pub metadata: Option<Value>,
+    pub metadata: Option<pii::Secret<Value>>,
```
Comment (SanchithHegde) [inline crates/router/src/compatibility/stripe/payment_intents/types.rs:66]:
```suggestion
    pub metadata: Option<pii::SecretSerdeValue>,
```

File: crates/router/src/compatibility/stripe/payment_intents/types.rs [hunk 3]
```diff
@@ -277,7 +277,7 @@ pub struct StripePaymentIntentResponse {
     pub customer: Option<String>,
     pub refunds: Option<Vec<refunds::RefundResponse>>,
     pub mandate_id: Option<String>,
-    pub metadata: Option<Value>,
+    pub metadata: Option<pii::Secret<Value>>,
```
Comment (SanchithHegde) [inline crates/router/src/compatibility/stripe/payment_intents/types.rs:280]:
```suggestion
    pub metadata: Option<pii::SecretSerdeValue>,
```

File: crates/router/src/connector/utils.rs [hunk 4]
```diff
@@ -105,8 +105,8 @@ impl<Flow, Request, Response> RouterData for types::RouterData<Flow, Request, Re
     where
         T: serde::de::DeserializeOwned,
     {
-        serde_json::from_value::<T>(self.get_connector_meta()?)
-            .into_report()
+        self.get_connector_meta()?
+            .parse_value("SecretSerdeValue")
```
Comment (SanchithHegde) [inline crates/router/src/connector/utils.rs:109]: The type name would be the type name of `T` here, isn't it?

Something like this:

```suggestion
            .parse_value(std::any::type_name::<T>())
```

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**